python/tests/test_highlevel.py

```diff
@@ -4261,3 +4261,23 @@ def test_macs_error(self):
             ValueError, match="macs output only supports single letter alleles"
         ):
             ts.to_macs()
+
+
+class TestTreeSequenceGetSite:
+    """
+    Tests for getting Site objects from a TreeSequence object
+    by specifying site id or position.
+    """
+
+    def get_example():
+        tables = tskit.TableCollection(sequence_length=10)
+        tables.sites.add_row(position=3, ancestral_state="A")
+        tables.sites.add_row(position=5, ancestral_state="C")
+        tables.sites.add_row(position=7, ancestral_state="G")
+        return tables.tree_sequence()
+
+    @pytest.mark.parametrize("position", [3.0, 5.0, 7.0])
```

Comment:
Also want to test different number types here, so something like
```
[3.0, 5.0, 7.0, 3, np.array([3], dtype=float)[0], np.array([3], dtype=int)[0], decimal.Decimal(3)]
```

Reply:
I have added a check that position must be an integer or floating point. Is that too narrow?

Reply:
Ahhh, why there are so many types of numbers?

Reply:
Checking types like this is usually an anti-pattern in python 

> I keep running into this error when trying to `git commit`:
> 
> ```
> flake8...................................................................Failed
> - hook id: flake8
> - exit code: 1
> 
> python/tests/test_highlevel.py:44:1: F401 'curses.ACS_NEQUAL' imported but unused
> python/tests/test_highlevel.py:52:1: F401 'python.tests.simplify.AncestorMap' imported but unused
> ```
> 
> So, I have been using `SKIP=flake8` for now.

@szhan Found that these lines had been added 
to his patch without his knowledge!
![9189283](https://user-images.githubusercontent.com/8552/170007855-0e3dc03b-d849-4c25-ba3b-aec26ab7daf9.jpg)